flink-state-backends/flink-statebackend-changelog/src/main/java/org/apache/flink/state/changelog/AbstractChangelogState.java

```diff
@@ -90,6 +90,10 @@ public void setCurrentNamespace(N namespace) {
         return delegatedState.getStateIncrementalVisitor(recommendedMaxNumberOfReturnedRecords);
     }
 
+    public void resetMetaDataWritten() {
```

Comment:
1. add javadoc?
2. **nit**: a more abstract name, something like `onChangelogTruncated`? (so that details aren't exposed, and it shouldn't be renamed if they change in the future)

Reply:
for function naming, I think `resetMetaDataWritten` is more descriptive for what it is doing than `onChangelogTruncated`. I would prefer to keep it as it is.

Reply:
I've changed `resetMetaDataWritten` to `resetWritingMetaFlag` (in this way, probably we do not need a java doc). WDYT?